cms/djangoapps/contentstore/management/commands/reindex_course.py

```diff
@@ -62,6 +62,40 @@ def _parse_course_key(self, raw_value):
 
         return result
 
+    def _setup(self):
+        """ Parses course key from string """
+        index_names = (CoursewareSearchIndexer.INDEX_NAME, CourseAboutSearchIndexer.INDEX_NAME)
+        for index_name in index_names:
+            try:
+                searcher = SearchEngine.get_search_engine(index_name)
+            except exceptions.ElasticsearchException as exc:
+                logging.exception('Search Engine error - %s', exc)
+                return
+
+            index_exists = searcher._es.indices.exists(index=index_name)  # pylint: disable=protected-access
+
+            index_mapping = searcher._es.indices.get_mapping(  # pylint: disable=protected-access
+                index=index_name,
+            ) if index_exists else {}
+
+            if index_exists and index_mapping:
+                return False
+        return True
+
+    def _get_active_course_keys(self, courses):
```

Comment:
maybe _filter_active_courses or _only_active(courses), it's a filter/drop rather than a get

Reply:
The thing about that naming is that it returns something different as well. You pass on a list of courses and you get back a list of course keys of the active courses of that list. Both `_filter_active_courses()` or `_only_active()` sounds like I'd filter the original list. This makes me think I should split the behavior to make it more clear. I'll refactor the block.

Reply:
I've refactored this in a more declarative way.